controller/contour/controller.go

```diff
@@ -119,71 +121,88 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // ensureContour ensures all necessary resources exist for the given contour.
 func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour) error {
+	var errs []error
 	// Before doing anything with the contour, ensure it has a finalizer
 	// so it can cleaned-up later.
 	if err := r.ensureFinalizer(ctx, contour); err != nil {
-		return fmt.Errorf("failed to finalize contour %s/%s: %w", contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to finalize contour %s/%s: %w", contour.Namespace, contour.Name, err))
 	}
 	if err := r.ensureNamespace(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
-			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
+			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err))
 	}
 	if err := r.ensureRBAC(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure rbac for contour %s/%s: %w", contour.Namespace, contour.Name, err)
+		errs = append(errs, fmt.Errorf("failed to ensure rbac for contour %s/%s: %w", contour.Namespace, contour.Name, err))
 	}
-	if err := r.ensureConfigMap(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure configmap for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureJob(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure job for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureDeployment(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure deployment for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureDaemonSet(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure daemonset for contour %s/%s: %w", contour.Namespace, contour.Name, err)
-	}
-	if err := r.ensureContourService(ctx, contour); err != nil {
-		return fmt.Errorf("failed to ensure service for contour %s/%s: %w", contour.Namespace, contour.Name, err)
+
+	var err error
+	deploy := &appsv1.Deployment{}
+	ds := &appsv1.DaemonSet{}
+	if len(errs) == 0 {
+		if err := r.ensureConfigMap(ctx, contour); err != nil {
+			errs = append(errs, fmt.Errorf("failed to ensure configmap for contour %s/%s: %w", contour.Namespace, contour.Name, err))
+		}
+		if err := r.ensureJob(ctx, contour); err != nil {
+			errs = append(errs, fmt.Errorf("failed to ensure job for contour %s/%s: %w", contour.Namespace, contour.Name, err))
+		}
+		deploy, err = r.ensureDeployment(ctx, contour)
```

Comment:
Can we move `err` into the inner scope and to the point of use?  
```suggestion
	deploy := &appsv1.Deployment{}
	ds := &appsv1.DaemonSet{}
	if len(errs) == 0 {
		if err := r.ensureConfigMap(ctx, contour); err != nil {
			errs = append(errs, fmt.Errorf("failed to ensure configmap for contour %s/%s: %w", contour.Namespace, contour.Name, err))
		}
		if err := r.ensureJob(ctx, contour); err != nil {
			errs = append(errs, fmt.Errorf("failed to ensure job for contour %s/%s: %w", contour.Namespace, contour.Name, err))
		}
		var err error
		deploy, err = r.ensureDeployment(ctx, contour)
```